object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -490,16 +501,21 @@ private void handleAssignFromRHS(
         LocalVariableNode mustCallChoiceParamLocal = getLocalPassedAsMustCallChoiceParam(rhs);
         if (mustCallChoiceParamLocal != null) {
           if (isVarInDefs(newDefs, mustCallChoiceParamLocal)) {
-            ImmutableSet<LocalVarWithTree> immutableSet =
+            LocalVarWithTree latestAssignmentPair =
+                getAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
+
+            ImmutableSet<LocalVarWithTree> setContainingLatestAssignmentPair =
                 getSetContainingAssignmentTreeOfVar(newDefs, mustCallChoiceParamLocal);
-            ImmutableSet<LocalVarWithTree> newImmutableSet =
-                FluentIterable.from(immutableSet)
+            ImmutableSet<LocalVarWithTree> newSetContainingLatestAssignmentPair =
+                FluentIterable.from(setContainingLatestAssignmentPair)
+                    .filter(Predicates.not(Predicates.equalTo(latestAssignmentPair)))
                     .append(
                         new LocalVarWithTree(
                             new LocalVariable((LocalVariableNode) lhs), node.getTree()))
                     .toSet();
```

Comment:
I still can't fully understand this code.  At this point, I think we are handling the case where (1) the `rhs` has a MCC variable `mustCallChoiceParamLocal` that is _already_ being tracked as part of a set, and also (2) `lhs` is also being tracked as part of a set.  (Question: do these have to be the same set?)  In such a case, we want to update the defs to still track a set with both `lhs` and `mustCallChoiceParamLocal`, but the tree for `lhs` should be this latest assignment.  Is this right?

If the above is right, please add a comment to this effect.

Reply:
> I still can't fully understand this code. At this point, I think we are handling the case where (1) the `rhs` has a MCC variable `mustCallChoiceParamLocal` that is _already_ being tracked as part of a set, and also (2) `lhs` is also being tracked as part of a set. (Question: do these have to be the same set?) 
>
Not necessarily. If these are in the same set, we will replace the previous assignment tree with the current assignment tree. Otherwise, `filter(Predicates.not(Predicates.equalTo(latestAssignmentPair)))` does nothing and then the currect assignment tree will be added to the set that contains `mustCallChoiceParamLocal`.
>In such a case, we want to update the defs to still track a set with both `lhs` and `mustCallChoiceParamLocal`, but the tree for `lhs` should be this latest assignment. Is this right?
>
It's true when these are in the same set.

Reply:
https://github.com/kelloggm/object-construction-checker/pull/282/commits/2dc62e3a4d37190f4881de233fdaa7fe334c8e53